sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,306 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
+ * @brief az_span_to_str copies a source span containing a string (not 0-terminated) to a
+ destination char buffer and appends the 0-terminating byte.
+
+ * The buffer referred to by destination must have a max_size that is at least 1 byte bigger
+ than the source az_span
+ *  the string in converts \p destination span to zero-terminated str. Content is copied to \p source
+ buffer and then \0 is
+ * addeed at the end. Then out_result will be created out of buffer
  *
- * @param self source span
- * @param out result span
- * @return AZ_NODISCARD az_span_to_uint64
+ * @param[in] destination A pointer to a buffer where the string should be copied
+ * @param[in] destination_max_size The maximum available space within the buffer referred to by destination.
+ * @param[in] source The az_span containing the not-0-terminated string
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK If /p source span's content is successfully copied to the destination.
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the /p destination buffer is too small to
+ copy the string and 0-terminate it
  */
-AZ_NODISCARD az_result az_span_to_uint64(az_span self, uint64_t* out);
+AZ_NODISCARD az_result
+az_span_to_str(char* destination, int32_t destination_max_size, az_span source);
+
+/******************************  SPAN PARSING */
 
 /**
- * @brief convert all the span content to uint32
+ * @brief az_span_to_uint64 parses an az_span containing ASCII digits into a uint64 number
  *
- * @param self source span
- * @param out result span
- * @return AZ_NODISCARD az_span_to_uint32
+ * @param[in] span The az_span containing the ASCII digits to be parsed.
+ * @param[in] out_number The pointer to the variable that is to receive the number
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_PARSER_UNEXPECTED_CHAR if a non-ASCII digit is found within the span.
  */
-AZ_NODISCARD az_result az_span_to_uint32(az_span self, uint32_t* out);
+
+AZ_NODISCARD az_result az_span_to_uint64(az_span span, uint64_t* out_number);
 
 /**
- * @brief move the content of span @b src to @b dst
+ * @brief az_span_to_uint32 parses an az_span containing ASCII digits into a uint32 number
  *
- * @param dst buffer where to copy span
- * @param src span to be copied
- * @param out span resulting after copy
- * @return AZ_NODISCARD az_span_move
+ * @param span The az_span containing the ASCII digits to be parsed.
+ * @param out_number The pointer to the variable that is to receive the number
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_PARSER_UNEXPECTED_CHAR if a non-ASCII digit
+ * is found within the span.
  */
-AZ_NODISCARD az_result az_span_copy(az_span dst, az_span src, az_span* out);
+AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number);
```

Comment:
Why don't we have other number sizes, including signed/unsigned flavors?

Reply:
We have only what we need. We don't add arbitrary functions in anticipation of customer needs. Code size is at a premium. Also, we don't want to document, test, & support functions that we don't need and we don't know if a customer needs either.